### DISCUSSION TITLE
Document core/supported/bundled resource types

### DISCUSSION THREAD

**Discussion by jamieklassen** [concourse/concourse#5460] [General]
Definitions:
* _maintaining_ a resource type means running automated tests, reviewing/merging PRs and shipping new versions to DockerHub.
* _bundling_ a resource type means that a version of the `concourse` binary is capable of running the resource type out-of-the-box without defining a `resource_type` for it. To reasonably _bundle_ a resource type, we have to also _maintain_ it.
* _supporting_ a resource type means committing to fixing bugs/CVEs. This should be the case for a subset of the _maintained_ resources, namely those that are _bundled_ in minor versions of Concourse for which we commit to regularly shipping patches.

Problems:
1. If I am a community contributor and I want to add a feature to a resource type, I'd like to know the set of types for which a full-time Concourse contributor will be reviewing my PR; i.e. which ones are maintained. I can reasonably figure this out by looking at https://resource-types.concourse-ci.org/ and seeing which cards are in the concourse org, but this heuristic isn't always going to work as we deprecate resource types. How can I get a clear picture of which resource types are maintained _right now_ so I know which ones will be good to contribute to?
2. If I am a concourse newbie, [the docs](https://concourse-ci.org/resource-types.html#resource-types) say
    > Concourse comes with a few "core" resource types

    but which ones are those? I want to know which resource types are _bundled_ in my concourse installation so that I can understand which resource `type:`s I can use out of the box. The answer to this comes from running `fly workers --details`, but this is not super discoverable. What can we do about that?
3. If I am a system architect or operator, I want know which resource types will be supported for the version line I have installed so that I can count on receiving bugfixes and security patches for my tenants who use them. The answer comes from the set of inputs to the `resource-types-images` job in the patching pipeline (e.g. https://ci.concourse-ci.org/teams/main/pipelines/release-5.5.x) for the version I have installed. This is not discoverable, and even if we documented the pipeline thing it would be a headache for most operators.

Possible solutions:
1. @ibagha had some great ideas, where "core" means _maintained_ as I've defined above (correct me if I'm wrong):
    > There are numerous ways we can display the core resources:
    > 1. Have a section at the top of the page for Concourse core resources (not as helpful IMO since that pushes the rest of catalog down on the page)
    > 2. Add a filter along with the ‘Popular’ and ‘A-Z’ filter for core resources
    > 3. Tag each resource card with pill that says ‘core’ or ‘core resource’
    > 4. Add a ‘core resources’ category if we reintroduce categories
    > 5. Add a new page as @jomsie mentioned
2. A note somewhere in https://resource-types.concourse-ci.org/ or https://concourse-ci.org/resource-types.html#resource-types (or both) saying "hey, run `fly workers --details` to see which resource types you can use out of the box" (or something like that).
3. Perhaps these should be declared in the enterprise docs, in a "compatibility-matrix"-type section like [this](https://docs.pivotal.io/p-concourse/v5/#bosh-concourse-compatibility), or maybe in the general [What is Concourse for VMware Tanzu](https://docs.pivotal.io/p-concourse/v5/what-is-concourse/) section.

**Comment by tremblaysimon** [comment]
It would be nice to be able to get the core resource type version used in a Concourse version too. Maybe with a verbose flavor of `fly workers --details` ?

**Reply by aoldershaw** [reply]
FWIW the resource type versions are emitted when using `fly workers --json`, e.g.

```
$ fly -t ci workers --json | jq '.[0].resource_types[] | {type: .type, version: .version}'
{
  "type": "bosh-io-release",
  "version": "1.1.1"
}
{
  "type": "bosh-io-stemcell",
  "version": "1.1.1"
}
{
  "type": "cf",
  "version": "1.1.3"
}
{
  "type": "docker-image",
  "version": "1.6.1"
}
{
  "type": "git",
  "version": "1.14.0"
}
{
  "type": "github-release",
  "version": "1.6.2"
}
{
  "type": "hg",
  "version": "1.2.4"
}
{
  "type": "mock",
  "version": "0.12.2"
}
{
  "type": "pool",
  "version": "1.2.1"
}
{
  "type": "registry-image",
  "version": "1.4.0"
}
{
  "type": "s3",
  "version": "1.1.2"
}
{
  "type": "semver",
  "version": "1.3.2"
}
{
  "type": "time",
  "version": "1.6.1"
}
{
  "type": "tracker",
  "version": "1.0.6"
}
```
